purchase_partner_approval/hooks.py

```diff
@@ -0,0 +1,16 @@
+from odoo import SUPERUSER_ID, api
+
+
+def approve_existing_partners_for_purchase(env):
+    """
+    To not cause disruption upon install,
+    set pre-existing Partners as Can Purchase
+    """
+    env["res.partner"].search([]).write(
```

Comment:
Do this with SQL:

```
UPDATE res_partner SET purchase_ok=True, candidate_purchase=True
```

although you should value if putting such defaults on the model. I think it's more frequent to allow by default than to block by default.

Reply:
Why is it bad to do this through the ORM?

Reply:
> although you should value if putting such defaults on the model. I think it's more frequent to allow by default than to block by default.

I'm not against it, but I didn't want to change the logic of the original module and wanted to get this fixed asap with as little fuss as possible.

Changed to SQL, which is much better performance-wise.

Reply:
I understand, but thinking twice, any new partner created in tests won't have these flags, and the behavior in this module will interfere in the tests, so I think it's saner to put the default and don't need the hook at all.

Reply:
done

Reply:
Though I'm not convinced this is the correct way to go about it, I feel like the module loses its purpose if you have to go out of your way to disapprove new contacts, as opposed to having to approve them.

Reply:
This is for data init only.
When you install, existing contacts are assumed as approved.
NEW contacts from there on would be subject to approval.